### PR TITLE
Ask user before enabling pruning

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/util/accountManager.ts [hunk 1]
```diff
@@ -164,6 +164,10 @@ async function createAccount(identityKeyPair: any) {
   // Enable typing indicators by default
   await Storage.put(SettingsKey.settingsTypingIndicator, false);
 
+  // opengroups pruning in ON by default on new accounts, but can change that from the settings
```
Comment (yougotwill) [inline ts/util/accountManager.ts:167]: Small typos should be "opengroups pruning is ON by default on new accounts, but you can change that from the settings"

File: _locales/en/messages.json [hunk 2]
```diff
@@ -137,6 +137,13 @@
   "typingIndicatorsSettingDescription": "See and share when messages are being typed (applies to all sessions).",
   "typingIndicatorsSettingTitle": "Typing Indicators",
   "zoomFactorSettingTitle": "Zoom Factor",
+  "pruneSettingTitle": "Prune Old Open Group Messages",
+  "pruneSettingDescription": "Prune messages older than 6 months from opengroups on start",
+  "pruningOpengroupDialogTitle": "Open group pruning",
+  "pruningOpengroupDialogMessage": "Pruning old open group messages improves performance. Enable pruning for open group messages older than 6 months?",
```
Comment (yougotwill) [inline _locales/en/messages.json:143]: Suggest we increase the line height of this text in the confirm modal it looks too squished. Inside `stylesheets/_session.scss` the css class starts at line 554. We should do this
```  
&-main-message {
    font-size: $session-font-md;
    line-height: 1.5;
}
```

File: _locales/en/messages.json [hunk 3]
```diff
@@ -137,6 +137,13 @@
   "typingIndicatorsSettingDescription": "See and share when messages are being typed (applies to all sessions).",
   "typingIndicatorsSettingTitle": "Typing Indicators",
   "zoomFactorSettingTitle": "Zoom Factor",
+  "pruneSettingTitle": "Prune Old Open Group Messages",
+  "pruneSettingDescription": "Prune messages older than 6 months from opengroups on start",
```
Comment (yougotwill) [inline _locales/en/messages.json:141]: I suggest replacing all `opengroups` text that is seen in the UI with `Open Groups` to be consistent.

File: ts/components/settings/section/CategoryAppearance.tsx [hunk 4]
```diff
@@ -119,6 +131,15 @@ export const SettingsCategoryAppearance = (props: { hasPassword: boolean | null
           description={window.i18n('audioMessageAutoplayDescription')}
           active={audioAutoPlay}
         />
+        <SessionToggleWithDescription
```
Comment (yougotwill) [inline ts/components/settings/section/CategoryAppearance.tsx:134]: I think the settings toggle for pruning Open Groups should be in the Privacy tab and not the Appearance tab but as you want.

File: ts/node/sql.ts [hunk 5]
```diff
@@ -3429,42 +3445,48 @@ function cleanUpOldOpengroups() {
...
+
+        if (messagesInConvoBefore >= maxMessagePerOpengroupConvo) {
+          const minute = 1000 * 60;
+          const userDefinedMonths = minute * 60 * 24 * 30 * pruneSetting;
+          const messagesTimestampToRemove = Date.now() - userDefinedMonths;
+          const countToRemove = assertGlobalInstance()
+            .prepare(
+              `SELECT count(*) from ${MESSAGES_TABLE} WHERE serverTimestamp <= $serverTimestamp AND conversationId = $conversationId;`
+            )
+            .get({ conversationId: convoId, serverTimestamp: Date.now() - userDefinedMonths })[
```
Comment (yougotwill) [inline ts/node/sql.ts:3461]: Can you not use the `messagesTimestampToRemove` here?

**Review by yougotwill — CHANGES_REQUESTED** [review]
Code looks code and running the branch in dev mode on my machine seems fine. Left a few minor comments.

Would you like me to do deep testing by running builds without users etc? Or will Emily do that in QA?

Nice work man 🚀

File: ts/components/settings/section/CategoryAppearance.tsx (hunk 4 above)
Comment (Bilb) [reply ts/components/settings/section/CategoryAppearance.tsx:134]: I like this

File: ts/node/sql.ts (hunk 5 above)
Comment (Bilb) [reply ts/node/sql.ts:3461]: right

**Pull request merged**